### DISCUSSION TITLE
D235 Video doorbell support

### DISCUSSION THREAD

**Discussion by phardy2410** [JurajNyiri/HomeAssistant-Tapo-Control#739] [Q&A] [answered]
Is there support for the new D235 Video doorbell yet? Looking to purchase in the next couple of weeks and will need the integration working.

Thanks for your continued hard work on this. It's a great thing.

**Comment by JurajNyiri** [comment]
Support will most likely be the same as for other doorbells.

**Reply by phardy2410** [reply]
Is that a yes?

Are other models supported? If so, then I will take it as a yes*

**Reply by JurajNyiri** [reply]
https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/issues/397
https://github.com/JurajNyiri/HomeAssistant-Tapo-Control?tab=readme-ov-file#troubleshooting--faq -> Supported models

I do not own it, but API is similar for each device.

**Reply by phardy2410** [reply]
Thanks. As you say it should be fine or I am sure you will be able to help if it doesn't work out the box

**Reply by JurajNyiri** [reply]
Yup, let me know.

**Reply by ssemones** [reply]
JurajNyiri I need assistance on how to ensure that the ports listed in the Readme are open on my network at my home location, is there a way to check those remotely or no? So to explain my setup I have a Optimum Cable Internet Gateway that feeds my Arris W31 Surfboard Mesh router, then I have a ethernet connection to a gigabit switch where I have multiple devices connected too including my Home Assistant Green, so I have been trying to get my Taco Video doorbells to show their video feeds in HA but the authentication fails each time I try to utilize the HACS integration Camera Control to configure the 2 video doorbells that I have. Now each of the devices I use to interface with HA use a VPN, so idk if that has any effect on the process but I have been at a loss trying to find anything specific describing a way(solution) to fix the issue. Thank you in advance.

**Comment by veeeps** [comment]
@JurajNyiri There is other threads discussing support of Tapo's doorbells and visibility of button push event. I got this doorbell and I added it to Home Assistant (TP-Link integration). A couple of entities was added automatically, but nothing relevant. Is there something I can look into to find out if we can dig more information via API or ONVIF? Especially I'm interested in having push button event. Just let me know how to test to get necessary information for you and I will. Thanks!

**Reply by noobipaul** [reply]
Did you manage to figure this out yet? I'm facing the same issue.

**Reply by JurajNyiri** [reply]
Form what was understood, it is not possible locally. Possibly with using 443mhz receiver.

**Reply by noobipaul** [reply]
Ok, thanks so much for your speedy response!

**Comment by PS1103** [comment]
Any news?

**Comment by geon106** [comment]
Any improvement on the D235? Just replaced my Ring Doorbell that would send motion events, push events and also send Live View / Last Recordings to Home Assistant

My D235 does none of this

**Comment by bkintanar** [comment]
I can't even add the integration of the D235 to Home Assistant, I was however add the chime the comes with the video doorbell.

**Comment by ssemones** [comment]
Mine works great in home assistant haven’t ran into any issues in quite
some time using the Tapo integration and making sure that all of my camera
accounts all have the same client account for access

**Comment by geon106** [comment]
Ive managed to add the D235 and C460, they have motion and person detection entities, but nothing gets send to HA to say there are any detection events. The logs are empty. But it does allow me to turn motion detection on and off. 

I gave up trying to get Last Recording or Live View of the cameras. It seems these devices do not allow rtsp unless hardwired for the D235 and the C460 doesn't at all even though I have the solar panel kit

**Reply by koendruyts** [reply]
> Ive managed to add the D235 and C460, they have motion and person detection entities, but nothing gets send to HA to say there are any detection events. The logs are empty. But it does allow me to turn motion detection on and off.
> 
> I gave up trying to get Last Recording or Live View of the cameras. It seems these devices do not allow rtsp unless hardwired for the D235 and the C460 doesn't at all even though I have the solar panel kit

I have the same issue....Is there a potential update comming to add the motion detection events via the integration?

**Comment by bkintanar** [comment]
@ssemones @geon106 are you using the D235 in hardwire mode or battery mode? I think HA needs the doorbell to be in alway-on mode (hardwire) to work properly.

**Reply by geon106** [reply]
> @ssemones @geon106 are you using the D235 in hardwire mode or battery mode? I think HA needs the doorbell to be in alway-on mode (hardwire) to work properly.

Mine is battery. I have no existing wires to use to plug it in. Its a shame Ring allows this but Tapo doesn't. Just spent a lot of money replacing 3x Ring stick up cams and a Ring 2nd gen doorbell and the HA side of things seems a step back.

Wish Tapo would improve this. In the meantime I wonder if there is any way I can wire a power bank to the doorbell so it thinks it is hardwired?

**Reply by bkintanar** [reply]
powerbank idea should be doable. i think

**Comment by geon106** [comment]
One way ive worked around no detection events or button press events is a super janky workaround.

I set an automation in the Tapo app to say if the doorbell button is pressed, turn on a Tapo P110 power socket. Then in HA, I use that socket turning on to trigger anything that is a doorbell press.

Does mean sacrificing a smart plug though

**Reply by bkintanar** [reply]
how were you able to add the doorbell to the integration? i cant add the host to it.

**Reply by geon106** [reply]
> how were you able to add the doorbell to the integration? i cant add the host to it.

I couldn't get the Doorbell nor Camera to add with the Tapo Control integration. It just gives an error when I enter their IP address.

So the Tapo app takes the event from the Doorbell which then triggers a smart plug to turn on, HA then has an automation using the Tapo Smart Home integration that says "if this smart plug turns on, then do x y z"

**Reply by JurajNyiri** [reply]
what error does it give?

**Reply by geon106** [reply]
> what error does it give?

Just "Connection Failed". Tried both D235 Doorbell and C460 camera.

I can add them using the TP Smart Home integration but that integration drains the batteries within a few hours and seems limited

**Reply by JurajNyiri** [reply]
No, there is a proper stack trace and log. Check home assistant log file.

**Comment by ssemones** [comment]
I apologize for the confusion everyone I have the D225 video doorbell, but
I was able to utilize the tplink smart home app for all my cameras and
smart home switches, the process involved me adding everything by ip
address and yes it is hardwired, I  then it asks for my Tapo/tplink
credentials I add those and it added it fine. I can watch the feed live and
see all the recorded events the only camera I am unable to view in home
assistant is the c425 which I can only view the feed in the Tapo app alone.

**Reply by bkintanar** [reply]
maybe because yours is hardwired makes it discoverable through the network or something. i dunno. I could add the ip address using the tplink smart home integration in HA, but when adding tplink email/password it says failed.

**Reply by JurajNyiri** [reply]
It will work soon in this integration (not core one), see my last update.

**Comment by ssemones** [comment]
I know some of the indoor cameras have a separate camera account that you
have to configure to allow for live viewing etc in Always on mode

**Comment by JurajNyiri** [comment]
Update: It is coming very soon (https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/pull/1007)

**Reply by bkintanar** [reply]
Will this work on battery mode as well, like an on demand option, or will this feature only benefit users using hardwired mode?

**Reply by JurajNyiri** [reply]
It will work on everything.

**Reply by bkintanar** [reply]
oh nice! thanks! I'm a developer as well, but mostly doing web application. I would love to tinker and be able to help.

**Reply by JurajNyiri** [reply]
It's all opensource, feel free to take a peak ;)

https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/pull/1007
https://github.com/JurajNyiri/pytapo/pull/149

**Comment by geon106** [comment]
![Screenshot_20250724_205746_Home Assistant](https://github.com/user-attachments/assets/c7553770-38e4-4bde-8da2-3a902c20068f)

I get an error whenever I try and add my D235 or C460 KIT devices using the Tapo Camera Control

**Reply by JurajNyiri** [reply]
Post logs.

**Reply by geon106** [reply]
> Post logs.

This is a good time to admit that im an idiot. How do I gather logs from the integration?

**Reply by JurajNyiri** [reply]
home-assistant.log in /config folder, or /config/logs via browser.

**Reply by geon106** [reply]
> home-assistant.log in /config folder, or /config/logs via browser.

![Screenshot_20250724_210835_Home Assistant](https://github.com/user-attachments/assets/ef0e4f3f-bd0f-4429-865d-1c2856452ac2)

Got it here in my root folder. Doesn't let me download it or copy/paste so will switch to my laptop and copy/paste the last 10 minutes where I attempted to add both devices

**Reply by JurajNyiri** [reply]
Ok, will wait for logs.

**Reply by geon106** [reply]
> Ok, will wait for logs.

Here we go
[home-assistant.log](https://github.com/user-attachments/files/21418584/home-assistant.log)

**Reply by JurajNyiri** [reply]
2025-07-24 12:48:08.022 ERROR (MainThread) [custom_components.tapo_control] Failed to establish a new connection

HA cannot connect to camera, you need to find out why. Are you entering correct IP? Can HA reach that IP?

**Reply by geon106** [reply]
> 2025-07-24 12:48:08.022 ERROR (MainThread) [custom_components.tapo_control] Failed to establish a new connection
> 
> HA cannot connect to camera, you need to find out why. Are you entering correct IP? Can HA reach that IP?

<img width="330" height="181" alt="image" src="https://github.com/user-attachments/assets/099199ba-beeb-4184-a396-c4c5acd9410f" />

I've double checked the IPs here as listed on my router. The Tapo app can see both devices and can be pinged. Does the integration rely on any external services? I do have Pi-Hole on my network if that could be causing a problem? Just in case I will disable that for a moment

**Reply by JurajNyiri** [reply]
ssh into home assistant and ping it, you can also do `curl https://192.168.0.57 -k --verbose` from HA, post output here

**Reply by geon106** [reply]
Right, disabled Pi-Hole and also restarted HA.

Added the Integration and went for the C460, it added straight away but in disabled state. This is because I disabled it in TP Smart Home integration due to that draining the battery. I deleted the camera within TP Smart Home so now it only shows in Tapo Camera Control and TP Router. After a few moments all the entities have come alive and seems to work a treat!

I shall add the D235 and monitor battery health, really hoping this integration doesn't reduce the battery life down to 20 hours haha.

Thank you for your assistance

**Reply by JurajNyiri** [reply]
Updates are already dynamic for battery devices vs main powered. You can configure update interval (and **a lot**) more via settings if you want it even slower.

**Reply by geon106** [reply]
The D235 gave the error again, but pressed next a few times in frustration then boom! Went to the next page for the Cloud password.

So either the restart of HA or disabling Pi-Hole did the trick. I will monitor it this weekend while I'm away and if it works great without battery drain, well @JurajNyiri drop me a link to buy you a coffee ;)

**Reply by JurajNyiri** [reply]
Haha. The link is in readme. Reach out to discord if you encounter more issues.

**Reply by bkintanar** [reply]
how do you disable Pi-Hole? I'm also having the same issue. I can't add the D235 using the integration. I'm using docker, and I'm not getting anything on ssh.

```sh
8022f2d8c83f:/config# curl https://192.168.1.33 -k --verbose
*   Trying 192.168.1.33:443...
* connect to 192.168.1.33 port 443 from 172.17.0.2 port 50226 failed: Connection refused
* Failed to connect to 192.168.1.33 port 443 after 75005 ms: Could not connect to server
* closing connection #0
curl: (7) Failed to connect to 192.168.1.33 port 443 after 75005 ms: Could not connect to server
```

**Reply by geon106** [reply]
> how do you disable Pi-Hole? I'm also having the same issue. I can't add the D235 using the integration. I'm using docker, and I'm not getting anything on ssh.
> 
> ```shell
> 8022f2d8c83f:/config# curl https://192.168.1.33 -k --verbose
> *   Trying 192.168.1.33:443...
> * connect to 192.168.1.33 port 443 from 172.17.0.2 port 50226 failed: Connection refused
> * Failed to connect to 192.168.1.33 port 443 after 75005 ms: Could not connect to server
> * closing connection #0
> curl: (7) Failed to connect to 192.168.1.33 port 443 after 75005 ms: Could not connect to server
> ```

![Screenshot_20250724_220141_Brave](https://github.com/user-attachments/assets/d8c50914-c368-498a-b872-355882657ff3)

Login to your Pi-Hole server, go to Disable Blocking. I disabled mine for 5 mins, restarted HA and then tried again and it worked immediately for my C460 and it worked for my D235 but I had to press Next after entering the IP address a few times

Unsure if in Docker it makes any difference as my Pi-Hole and HA are each installed on their own dedicated servers

**Reply by bkintanar** [reply]
I fixed mine by raping the next button, it turns out that since my video doorbell is battery powered, then it needs to be woken up for this to work. and a single click on the next button doesn't wake the device so it can't see the ip address.

**Comment by bkintanar** [comment]
OK my issue is now fixed with the latest 7.0.0 release. I can see the stream now. Just make sure to disable the "Use Stream from Home Assistant", under the:

Settings -> Devices & Services -> Tapo: Cameras Control -> THE COG -> Configure Device -> uncheck Use Stream from Home Assistant (restart required)

Thank me later.

Thank you so much @JurajNyiri

**Comment by JurajNyiri** [comment, marked as answer]
Now available in 7.0.0 release.

**Reply by Only-4-Info** [reply]
Great work. Thanks.
I am on « HA Green » hardware. Do you know when and in which core version it will be packaged ?
(I don’t know if I can install your 7.0.0 GitHub version on the HA Green)
Many thanks. Regards

**Reply by JurajNyiri** [reply]
Yes you can follow readme instructions.

**Comment by boereworswizard** [comment]
Has anyone gotten 2-way talk to work?

**Comment by howels** [comment]
Unfortunately D235 firmware 1.2.1 appears to have broken access to the doorbell:

```
Failed setup, will retry: HTTPSConnectionPool(host='x.x.x.x', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] ssl/tls alert handshake failure (_ssl.c:1032)')))
```

**Comment by mondychan** [comment]
Fw 1.2.1, hardwired, cloud account and also the local one gives error
 try_connect_all failed
